File: pinpoint/lib/model/admin/admin_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pinpoint/model/address/address.dart';
import 'package:pinpoint/model/batch/batch_list_response.dart';
import 'package:pinpoint/model/notice/notice.dart';

part 'admin_response.g.dart';


@JsonSerializable()
class AdminResponse {
  final String id;
  final String email;
  final String phone;
  final String? name;
  final String role;
  final Address? address;
  final String instituteId;
  final List<BatchListResponse> batches;
  final List<NoticeDto> notices;
  final String createdAt;
  final String updatedAt;
  @JsonKey(name: 'verified')
  final bool isVerified;

  AdminResponse({
    required this.id,
    required this.email,
    required this.phone,
    this.name,
    required this.role,
    this.address,
    required this.instituteId,
    required this.batches,
    required this.notices,
    required this.createdAt,
    required this.updatedAt,
    required this.isVerified,
  });

  factory AdminResponse.fromJson(Map<String, dynamic> json) =>
      _$AdminResponseFromJson(json);

  Map<String, dynamic> toJson() => _$AdminResponseToJson(this);
}

File: pinpointB/src/main/java/com/nxquar/pinpoint/DTO/AdminResponse.java
package com.nxquar.pinpoint.DTO;

import com.nxquar.pinpoint.DTO.branch.BatchListResponse;
import com.nxquar.pinpoint.Model.Address;
import com.nxquar.pinpoint.Model.Notice;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AdminResponse {
    private UUID id;
    private String email;
    private String phone;
    private String name;
    private String role;
    private Address address;
    private UUID instituteId;
    private List<BatchListResponse> batches;
    private List<Notice> notices;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private boolean isVerified;
}
